include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1012 @@
+        , plus_to_space_(plus_to_space)
+    {}
+#endif
+
+    /** Returns an iterator to the beginning
+
+        Returns an iterator to the first decoded character of the string
+
+        @return const_iterator to the first character
+    */
+    BOOST_URL_DECL
+    const_iterator
+    begin() const noexcept;
+
+    /// @copydoc begin()
+    BOOST_URL_DECL
+    const_iterator
+    cbegin() const noexcept;
+
+    /** Returns an iterator to the end
+
+        Returns an iterator to one past the last decoded character of the string.
+
+        This character acts as a placeholder, attempting to access it
+        results in undefined behavior.
+
+        @return const_iterator to the character following the last character
+    */
+    BOOST_URL_DECL
+    const_iterator
+    end() const noexcept;
+
+    /// @copydoc end()
+    BOOST_URL_DECL
+    const_iterator
+    cend() const noexcept;
+
+    /** Accesses the first character
+
+        Returns the decoded first character in the string.
+
+        The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+     */
+    BOOST_URL_DECL
+    const_reference
+    front() const;
+
+    /** Accesses the last character
+
+        Returns the last decoded character in the string
+
+        The behavior is undefined if `empty() == true`.
+
+        @return reference to the first character
+     */
+    BOOST_URL_DECL
+    const_reference
+    back() const;
+
+    /** Returns a pointer to the first encoded character of a string
+
+        Returns a pointer to the underlying character array.
+
+        The pointer is such that the range `[data(); data() + encoded_size())`
+        is valid and the values in it correspond to the values of the
+        string.
+
+        @note Unlike `std::basic_string::data()` and string literals,
+        `data()` may return a pointer to a buffer that is not
+        null-terminated. Therefore, it is typically a mistake to pass
+        data() to a routine that takes just a const char* and expects
+        a null-terminated string.
+
+        @return A pointer to the underlying character array.
+     */
+    BOOST_CONSTEXPR
+    const char*
+    encoded_data() const
+    {
+        return p_;
+    }
+
+    /** Returns the number of char elements in the string
+
+        @return The number of char elements in the string
+    */
+    BOOST_CONSTEXPR
+    size_type
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /// @copydoc size()
+    BOOST_CONSTEXPR
+    size_type
+    length() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Returns the number of encoded char elements in the string
+
+        @return The number of encoded char elements in the string
+    */
+    BOOST_CONSTEXPR
+    size_type
+    encoded_size() const noexcept
+    {
+        return n_;
+    }
+
+    /// @copydoc encoded_size()
+    BOOST_CONSTEXPR
+    size_type
+    encoded_length() const noexcept
```

Comment:
Too much try-hard here why are there so many functions which do the same thing, a lot of repetition, a lot of busy work, a lot of boilerplate, a lot of.. there's just TOO MANY NOTES
https://www.youtube.com/watch?v=dCud8H7z7vU

Reply:
I'm Mozart?

Reply:
Sure :)